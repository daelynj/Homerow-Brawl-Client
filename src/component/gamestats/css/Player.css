.Player {
  float: left;
  padding-top: 10px;
  padding-left: 10px;
}
