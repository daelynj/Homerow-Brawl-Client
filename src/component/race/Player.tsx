import * as React from "react";
import octane from "./images/octane.png";

interface Props {
  position: number;
  name: string;
}

export const Player = (props: Props) => (
  <div>
    {props.name}
    <img
      style={{
        paddingLeft: props.position + "%"
      }}
      src={octane}
      alt="Octane"
    />
  </div>
);
