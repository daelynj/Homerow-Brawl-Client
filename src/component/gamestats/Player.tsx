import * as React from "react";
import "./css/Player.css";

interface Props {
  name: string;
  WPM: number;
  accuracy: number;
  wordsTyped: number;
  time: number;
  mistakes: number;
}

export const Player = (props: Props) => (
  <span className="Player">
    <div>{"name: " + props.name}</div>
    <div>{"WPM: " + props.WPM}</div>
    <div>{"accuracy: " + props.accuracy + "%"}</div>
    <div>{"words typed: " + props.wordsTyped}</div>
    <div>{"time: " + props.time + " seconds"}</div>
    <div>{"mistakes: " + props.mistakes}</div>
  </span>
);
