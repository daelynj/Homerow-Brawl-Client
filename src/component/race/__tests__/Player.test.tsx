import * as React from "react";
import { Player } from "../Player";
import renderer from "react-test-renderer";

describe("Player", () => {
  it("renders the described Player", () => {
    const tree = renderer.create(<Player position={10} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
