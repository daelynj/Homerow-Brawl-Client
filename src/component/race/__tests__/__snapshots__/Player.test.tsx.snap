// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders the described Player 1`] = `
<div
  style={
    Object {
      "paddingLeft": "10%",
      "textAlign": "left",
    }
  }
>
  <img
    alt="Octane"
    src="octane.png"
  />
  <div>
    octane
  </div>
</div>
`;
